### ISSUE TITLE
Bulk Add Contents to Project

### ISSUE THREAD

**Issue by mnelson4** [mnelson4/printmyblog#106]
Daniel also suggested having a way to add all posts, or all pages, etc to a project.
The use-case is when you have a project like "WordPress Wisdom", and you're always writing new content for it.
It would be great if it were easy to add new stuff to it, and order it. But we'd also want to detect duplicate content and avoid re-adding it... so we might also want a button to "Add All, Avoid Dups"

This will make creating massive project easier... which will expose the fact that PMB doesn't yet handle massive projects.

**Comment by mnelson4** [comment]
Sedlex also suggested this, but they wanted a bit more I documented on #120

**Comment by mnelson4** [comment]
Got this requested again (actually they asked for a refund without it, which stung.)
They suggested

> When building out the PDF - we have hundreds of posts to add. It just isn't an interface that the client can use. Perhaps if there was a way to drop in everything from a custom post type or custom category type.

**Comment by mnelson4** [comment]
Note to self: Lindsey Rampton said they'd like to try this out when it's ready (check my hotmail inbox)

**Comment by ericjung** [comment]
>  It just isn't an interface that the client can use

Simply being able to move **_all_** posts to the output would be a huge improvement. Dragging and dropping 300+ items right now is painful, especially with such tiny widgets for drag-and-drop areas.

**Comment by mnelson4** [comment]
Ok I have a modified version that:
* lets you select multiple posts (by using control or shift + clicking on them)
* then drag all the selected posts to wherever you see best
It's on branch **FET/multidrag** which you can download here: https://github.com/mnelson4/printmyblog/archive/refs/heads/FET/multidrag.zip (just download it, then use WordPress' "Add New" plugin page, select "Upload" and upload it and activate it, then deactivate the other version of PMB). 

@ericjung if you get a chance to try it, here are my thoughts:

* would a button to "select all" be helpful? (Currently you can click on one item, hold shift, then click on the last, to select them all..)
* if so, should it also automatically load all posts matching the search criteria (currently PMB just loads 50 then stops until you click to load more)
* any other thoughts on how to make using it more intuitive and easy?

**Comment by ericjung** [comment]
Hi @mnelson4. Great work -- I've already tried it.

> would a button to "select all" be helpful?

Absolutely

> if so, should it also automatically load all posts matching the search criteria 

Yes, definitely. I think a common usage pattern, and the one I'm certainly going to use, is this:

1. Generate a PDF with all of my posts the first time, using very limited criteria (certainly not dates). Print.
2. Once per month or once per X days, generate a new PDF with search criteria leveraging dates. This will include only new posts since step 1. Print and append to the document from step 1.
3. Repeat step 2 but include only new posts created after step 2.
4. Repeat Ad infinitum.

> any other thoughts on how to make using it more intuitive and easy?

Sure! Look at this screenshot (I've blurred the titles for privacy):

![image](https://user-images.githubusercontent.com/5875610/124679907-90b7a000-de82-11eb-888f-cf223226c37c.png)

You can see that sometimes -- most of the time for me -- the *Project Content* block wraps. This makes the whole drag-and-drop experience pretty awful. I know you did not intend that. I would at least put the *Search and Apply Filters* block in its own section, full-width. Then try to put only *Search Results* and *Project Content* side-by-side [something like this](https://github.com/atlassian/react-beautiful-dnd)

Reordering of the final list is not a big deal to me since I always want to order by date, but I can see how other people might want that.

**Comment by mnelson4** [comment]
Thanks for the feedback, that's all very helpful.
I've created an issue to followup on the wrapping issue: https://github.com/mnelson4/printmyblog/issues/256.
The rest of I'll do on this issue

**Comment by mnelson4** [comment]
Ok I have a pretty big update if you want to take a peek @ericjung.
Here's what I have now: 
https://i.imgur.com/iqqMH3l.gif

What's new:
* filters initially hidden (to simplify page)
* search bar still visible
* text saying to use CTRL or SHIFT to select multiple items
* button to add and remove selected items
* buttons to move items up or down
* button to "load all", "select all" and "deselect all"

Try it by downloading this new branch: https://github.com/mnelson4/printmyblog/archive/refs/heads/FET/content-edit-improvements.zip

**Comment by ericjung** [comment]
@mnelson4  great work! This is brainless and painless.

**Comment by mnelson4** [comment]
Thanks! I'm still trying to get some more usability feedback before I release it (I'll need to rewrite some documentation too after it's released)

**Comment by mnelson4** [comment]
* [ ] after bulk moving, selecting other content goes weird. See https://github.com/SortableJS/Sortable/issues/1827

**Comment by mnelson4** [comment]
* [ ] apply Ben's easy suggestions
* [ ] update documentation and videos

**Comment by mnelson4** [comment]
Ok this has been merged into master and I've updated the documentation https://printmy.blog/user-guide/getting-started/6-choose-project-content
